app/scripts/modules/core/cancelModal/cancelModal.controller.ts

```diff
@@ -0,0 +1,44 @@
+import {module} from 'angular';
+import {IModalServiceInstance} from 'angular-ui-bootstrap';
+import IScope = ng.IScope;
+
+require('./cancel.less');
+
+interface ICancelModalScope extends IScope {
+  state: any;
+  params: any;
+  errorMessage: string;
+}
+
+export class CancelModalCtrl {
+
+  constructor(public $scope: ICancelModalScope, private $uibModalInstance: IModalServiceInstance, private params: any) {
+    this.$scope.params = params;
+
+    this.$scope.state = {
+      submitting: false
+    };
+  }
+
+  formDisabled = () => this.$scope.state.submitting;
+
+  showError(exception: string): void {
+    this.$scope.state.error = true;
+    this.$scope.state.submitting = false;
+    this.$scope.errorMessage = exception;
+  }
+
+  confirm(): void {
+    if (!this.formDisabled()) {
```

Comment:
I meant that you should also have a check here for whether or not the form is already being submitted in your if statement.  Currently you are relying on the angular digest cycle to disable the button in your modal and you can't do this in a deterministic way.